roles/update/tasks/update_run.yml

```diff
@@ -0,0 +1,196 @@
+
+- name: Wait for ovn update completion on data plane
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    api_version: core.openstack.org/v1beta1
+    kind: OpenStackVersion
+    namespace: "{{ cifmw_update_namespace }}"
+  register: _openstackversion_info
+  retries: 20
+  delay: 15
+  until:
+    - _openstackversion_info.resources[0].status.conditions is defined
+    - >-
+      _openstackversion_info.resources[0].status.conditions |
+      selectattr("type", "equalto", "MinorUpdateOVNDataplane") |
+      selectattr("status", "equalto", "True") | length > 0
+
+- name: Wait for control plane update completion
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    api_version: core.openstack.org/v1beta1
+    kind: OpenStackVersion
+    namespace: "{{ cifmw_update_namespace }}"
+  register: _openstackversion_info
+  retries: 20
+  delay: 15
+  until:
+    - _openstackversion_info.resources[0].status.conditions is defined
+    - >-
+      _openstackversion_info.resources[0].status.conditions |
+      selectattr("type", "equalto", "MinorUpdateControlplane") |
+      selectattr("status", "equalto", "True") | length > 0
+
+- name: Fetch NodeSets for update OpenStackDataPlaneDeployment
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path | default(ansible_env.PATH) }}"
+  ansible.builtin.shell: >-
+    set -o pipefail;
+    oc -n {{ cifmw_update_namespace }}
+    get openstackdataplanenodeset -o name
+    | awk -F'/' '{print $2}'
+  register: cifmw_update_node_sets
+  changed_when: false
+
+- name: Create the OpenStackDataPlaneDeployment CR used for ovn udpate
+  ansible.builtin.copy:
+    dest: "{{ cifmw_update_artifacts_basedir }}/edpm_update.yaml"
+    content: "{{ _content | to_nice_yaml }}"
+    mode: "0644"
+  vars:
+    _content:
+      apiVersion: dataplane.openstack.org/v1beta1
+      kind: OpenStackDataPlaneDeployment
+      metadata:
+        name: edpm-update
+        namespace: "{{ cifmw_update_namespace }}"
+      spec:
+        nodeSets: "{{ cifmw_update_node_sets.stdout
+        | split('\n')
+        | map('trim')
+        | reject('equalto', '')
+        | list
+        }}"
+        servicesOverride:
+          - update
+
+- name: Create the OpenStackDataPlaneDeployment CR to trigger OpenStackDataPlaneDeployment update
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path | default(ansible_env.PATH) }}"
+  ansible.builtin.command: >-
+    oc -n {{ cifmw_update_namespace }}
+    create -f {{ cifmw_update_artifacts_basedir }}/edpm_update.yaml
+
+- name: Wait for update completion on data plane
```

Comment:
should there be a compute nodes reboot here ? is there a reboot handling of the computes nodes going on later in the existing workflow?

Reply:
Reboot is tested later in existing workflow: https://github.com/openstack-k8s-operators/ci-framework/blob/b7e23c7bb7687c22120afd865dce420adc0d6e5b/roles/update/tasks/main.yml#L78